pentaho-js/src/main/javascript/cdf/components/TableComponent.js

```diff
@@ -492,200 +522,246 @@ define([
      * called, or false otherwise.
      */
     handleAddIns: function(dataTable, td, $td, rowIdx, colIdx) {
-      var cd = this.chartDefinition,
-          colType = cd.colTypes[colIdx],     
-          state = {},
-          target = $td,
-          results = this.rawData,
-          addIn = this.getAddIn("colType", colType);
+      var cd = this.chartDefinition;
+      var colType = cd.colTypes[colIdx];
+      var state = {};
+      var target = $td;
+      var results = this.rawData;
+
+      var addIn = this.getAddIn("colType", colType);
       if(!addIn) {
         return false;
       }
+
       try {
         if(!(target.parents('tbody').length)) {
           return;
-        } else if(target.get(0).tagName != 'TD') {
+        } else if(target.get(0).tagName !== 'TD') {
           target = target.closest('td');
         }
+
         state.rawData = results;
         state.tableData = dataTable.fnGetData();
         state.colIdx = colIdx;
         state.rowIdx = rowIdx;
         state.series = results.resultset[state.rowIdx][0];
         state.category = results.metadata[state.colIdx].colName;
         state.value = results.resultset[state.rowIdx][state.colIdx];
+
         if(cd.colFormats) {
           state.colFormat = cd.colFormats[state.colIdx];
         }
+
         state.target = target;
         addIn.call(td, state, this.getAddInOptions("colType", addIn.getName()));
+
         return true;
-      } catch(e) {
-        Logger.exception(e);
+      } catch(ex) {
+        Logger.exception(ex);
+
         return false;
       }
     },
 
     processTableComponentResponse: function(json) {
-      var myself = this,
-          cd = myself.chartDefinition,
-          extraOptions = {};
+      var myself = this;
+      var chartDefinition = this.chartDefinition;
+      var tablePlaceholder = this.ph;
+
+      if (this.dataTable != null) {
```

Comment:
To preserve backward compatibility the most, probably this should be done after the `trigger` below...